src/main/java/org/folio/innreach/domain/service/impl/PatronTypeMappingServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package org.folio.innreach.domain.service.impl;
+
+import org.folio.innreach.domain.entity.CentralServer;
+import org.folio.innreach.domain.exception.EntityNotFoundException;
+import org.folio.innreach.dto.PatronTypeMappingsDTO;
+
+import lombok.RequiredArgsConstructor;
+import org.folio.innreach.domain.entity.PatronTypeMapping;
+import org.folio.innreach.domain.service.PatronTypeMappingService;
+import org.folio.innreach.mapper.PatronTypeMappingMapper;
+import org.folio.innreach.repository.PatronTypeMappingRepository;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import static org.folio.innreach.domain.service.impl.ServiceUtils.centralServerRef;
+import static org.folio.innreach.domain.service.impl.ServiceUtils.initId;
+import static org.folio.innreach.domain.service.impl.ServiceUtils.mergeAndSave;
+
+@RequiredArgsConstructor
+@Service
+@Transactional
+public class PatronTypeMappingServiceImpl implements PatronTypeMappingService {
+
+  private final PatronTypeMappingRepository repository;
+  private final PatronTypeMappingMapper mapper;
+
+  @Override
+  public PatronTypeMappingsDTO getAll(UUID centralServerId, int offset, int limit) {
+    var example = mappingExampleWithServerId(centralServerId);
+
+    Page<PatronTypeMapping> mappings = repository.findAll(example, PageRequest.of(offset, limit));
+
+    return mapper.toDTOCollection(mappings);
+  }
+
+  @Override
+  public PatronTypeMappingsDTO updateAll(UUID centralServerId, PatronTypeMappingsDTO patronTypeMappingsDTO) {
+    var stored = repository.findAll(mappingExampleWithServerId(centralServerId));
+
+    var incoming = mapper.toEntities(patronTypeMappingsDTO.getPatronTypeMappings());
+    var csRef = centralServerRef(centralServerId);
+    incoming.forEach(setCentralServerRef(csRef).andThen(initId()));
```

Comment:
`id` is autogenerated since it has generation strategy = GenerationType.AUTO, I think there is no need to `initId`

Reply:
We need to call `initId()` because in the method `mergeAndSave(incoming, stored, repository, this::copyData)` entities are being compared by their ids.